shared-libs/cht-datasource/src/local/libs/core.ts

```diff
@@ -0,0 +1,11 @@
+import { Nullable } from '../../libs/core';
+import { InvalidArgumentError } from '../../libs/error';
+
+/** @internal */
+export const validateCursor = (cursor: Nullable<string>): number => {
```

Comment:
We initially started with `skip` as an integer parameter like `limit` but we realized that `skip` might sometimes have ambiguous meaning in terms of the validation logic we have in the code. For eg: a user might request 100 items with a skip initially as 0 and ideally, we would return 100 items but what if some of the items are invalid in the 1-100 range. We would need to add some from the 101-200 range to fill up the 100 items quota. Additionally, that would mean when we return the next `skip` value to the caller the value would end up being larger than 100. This might confuse the users as "I'm requesting 100 items and there are 100 items but the next `skip` value is `105`"". So, we decided to abstract the implementation from the user and use a `cursor` instead. This would be helpful if we decide to refactor and not use `skip` internally to query the database at all.
More details [here](https://github.com/medic/cht-core/pull/9266#discussion_r1692002639).

Reply:
Not really. It still means `skip` but with the implementation abstracted.